src/Bicep.Core.Samples/Files/Completions/declarations.json

```diff
@@ -939,6 +1478,17 @@
     "textEdit": {
       "range": {},
       "newText": "resource ${1:recoveryServiceVault} 'Microsoft.RecoveryServices/vaults@2021-01-01' = {\n  name: ${2:'name'}\n  location: resourceGroup().location\n  sku: {\n    name: ${3|'RS0','Standard'|}\n    tier: ${4:'Standard'}\n  }\n}\n"
+    },
+    "command": {
+      "command": "bicep.Telemetry",
+      "arguments": [
+        {
+          "EventName": "TopLevelDeclarationSnippetInsertion",
```

Comment:
nit: Rather long telemetry event name.  Consider shortening?

Is it possible that later snippet insertions would not be top-level or declaration?  I.e. SnippetInsertion not good enough?

Reply:
We want to add telemetry for resource and module body snippet insertions as well. We could also do something like - SnippetInsertion/TopLevel
SnippetInsertion/ResourceBody
SnippetInsertion/ModuleBody

I think it would show up as vscode-bicep/SnippetInsertion/TopLevel

@MarcusFelling , do you have any suggestions?

Reply:
Not a big deal, you just have to deal with whatever names you end up with in queries.